### DISCUSSION TITLE
Heater Cooler question

### DISCUSSION THREAD

**Discussion by Faisalthe01** [newt10/homebridge-broadlink-rm-enhanced#7] [Q&A]
Hey mate, great job with the Heater-Cooler Accessory. 

I have one question after reading your notes. 

If my air-con remote send a different hex code for a single Temp with a FanSpeed100 or a FanSpeed50 and then i record both hex codes in my config and then when the FanSpeed is 100 and i change the temperature to 17 it will send the hex code for Temp 17 with FanSpeed100 and if the FanSpeed is 50 and i change the temperature to 17 it will send the hex code for Temp 17 with FanSpeed50, correct?

**Comment by newt10** [comment]
Yes, it will use the current state of fanspeed (as well as oscillation) and send the appropriate code for changing temperature while keeping other attributes as they were. Let me know if you have any issues in setting up the config file. Please feel free to share your config file along here for additional help

**Comment by Faisalthe01** [comment]
thanks for the reply. So i am already using Heater-Cooler that I created and is based on the air-con accessory from the original plugin. But the problem is that when I change temperature or fanspeed from my Air-Con remote it sends a combination of different hex code. if the fanspeed is 2 and temperature is 20 it will send a different hex code similarly if the fanspeed is 1 and temperature is 20 it will send a different hex code but it still be a single hex code. I was thinking how can i maintain the current fanspeed is i only change the temperature of maintain the current temperature if i only  change the fanspeed

**Comment by newt10** [comment]
This plugin solves this problem. The hierarchical hex code structure for heater-cooler accessory in the config will allow you to provide a single hex code for temp 20 + fanspeed 1 and a different single hex code for temp 20 + fanspeed2

`data: {
...,
temperatureCodes: {
20: {
rotationSpeed50: "hex code 1",
rotationSpeed100: "hex code 2"
}
}`

Please check the sample config files and [help docs](https://github.com/newt10/homebridge-broadlink-rm-enhanced/blob/master/docs/heater-cooler.md#how-to-set-up-configjson) as it answers your question.
